ddtrace/internal/writer.py

```diff
@@ -243,6 +278,20 @@ def flush_queue(self):
         encoded = self._encoder.join_encoded(enc_traces)
         self._send_payload(encoded, len(enc_traces))
 
+        if self._report_metrics:
+            # Note that we cannot use the batching functionality of dogstatsd because
+            # it's not thread-safe.
+            # https://github.com/DataDog/datadogpy/issues/439
```

Comment:
let's talk about this, we should be able to figure out some way to write a batched payload. I mean, we are batching the data ourselves anyways

Reply:
We should just fix it upstream

Reply:
In fact most of the ideas in #1143 should be proposed upstream since these kinds of issues are not at all unique to us.